maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/BuilderCommon.java

```diff
@@ -141,35 +142,36 @@ public MavenExecutionPlan resolveBuildPlan( MavenSession session, MavenProject p
             final Set<Plugin> unsafePlugins = executionPlan.getNonThreadSafePlugins();
             if ( !unsafePlugins.isEmpty() )
             {
-                logger.warn( "*****************************************************************" );
-                logger.warn( "* Your build is requesting parallel execution, but project      *" );
-                logger.warn( "* contains the following plugin(s) that have goals not marked   *" );
-                logger.warn( "* as @threadSafe to support parallel building.                  *" );
-                logger.warn( "* While this /may/ work fine, please look for plugin updates    *" );
-                logger.warn( "* and/or request plugins be made thread-safe.                   *" );
-                logger.warn( "* If reporting an issue, report it against the plugin in        *" );
-                logger.warn( "* question, not against maven-core                              *" );
-                logger.warn( "*****************************************************************" );
+                for ( String s : MultilineMessageHelper.format(
+                        "Your build is requesting parallel execution, but this project contains the following "
+                                + "plugin(s) that have goals not marked as @threadSafe to support parallel execution.",
```

Comment:
Nit: `@threadSafe` suggests the (old) Javadoc notation. Wouldn't it make more sense to do either `@ThreadSafe` or just "not marked as thread-safe" nowadays?

Reply:
That is true, the Javadoc tag is gone and we don't have a separate Java annotation for "thread-safe". You proposal is correct.

Reply:
Done.